cmd/goProbe/cmd/root.go

```diff
@@ -0,0 +1,362 @@
+// Package cmd contains the goProbe command line interface implementation
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/els0r/goProbe/v4/pkg/api/server"
+	"github.com/els0r/goProbe/v4/pkg/capture"
+	"github.com/els0r/goProbe/v4/pkg/conf"
+	"github.com/els0r/goProbe/v4/pkg/version"
+	"github.com/els0r/telemetry/logging"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"golang.org/x/time/rate"
+
+	gpconf "github.com/els0r/goProbe/v4/cmd/goProbe/config"
+	gpserver "github.com/els0r/goProbe/v4/pkg/api/goprobe/server"
+)
+
+const shutdownGracePeriod = 30 * time.Second
+
+var (
+	defaultRequestDurationHistogramBins = []float64{0.01, 0.05, 0.1, 0.25, 1, 5, 10, 30, 60, 300}
+)
+
+func Execute() error {
+	rootCmd, err := newRootCmd(run)
+	if err != nil {
+		return err
+	}
+
+	rootCmd.AddCommand(newVersionCmd())
+
+	return rootCmd.Execute()
+}
+
+// runFunc is the type of the function that is called when the root command is executed. It's defined
+// mainly for testing purposes
+type runFunc func(ctx context.Context, cfg *gpconf.Config) error
+
+func newRootCmd(run runFunc) (*cobra.Command, error) {
+	cfg := &gpconf.Config{}
+
+	rootCmd := &cobra.Command{
+		Use:   "goProbe",
```

Comment:
More of a nit, but IMHO we should move away from CC ~here~ (everywhere) and be more idiomatic for the tool naming:
`goProbe` -> `goprobe`
`goQuery` -> `goquery`

I've noticed that in a different context, but when you look at Linux OSes, convention is to use all-lower names for command line tools (anything else is "snowflaky"). WDYT?